include/aws/common/byte_buf.h

```diff
@@ -113,11 +113,19 @@ bool aws_array_eq_c_str_ignore_case(const void *array, size_t array_len, const c
 AWS_COMMON_API
 int aws_byte_buf_init(struct aws_byte_buf *buf, struct aws_allocator *allocator, size_t capacity);
 
+/**
+ * Initializes *dest by copying all the fields from *src.
```

Comment:
Better comment here: what are the requirements? When can it fail? it doesn't just copy the fields, it allocates a new backing array.